swift/ios-shoppe-demo/ServiceManagers/FullStoryManager.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  FullStoryManager.swift
+//  ios-shoppe-demo
+//
+//  Created 5/7/20.
+//  Copyright © 2020 FullStory All rights reserved.
+//
+
+import Foundation
+import FullStory
+
+/**
+ FullStory Manager is a singleton used to manage api calls to FullStory.
+
+    - create events
+    - identify a user masking and unmasking elements
+ */
+class FullStoryManager {
+    static var shared = FullStoryManager()
+
+    enum Event: String {
+        case browsing
+        case itemSelected
+        case viewCart
+        case addToCart
+        case checkout
+        case removeFromCart
+        case crash
+    }
+
+    enum LogLevel {
+        case assert, error, warning, info, debug
+    }
+
+    enum FSStatus: String {
+        case exclude = "fs-exclude"
+        case mask = "fs-mask"
+        case unmask = "fs-unmask"
+    }
+
+    var currentEvent: Event = .browsing
+
+    func createFSEvent(event: Event? = nil) {
```

Comment:
This has a side-effect of also logging.  That's not immediately clear from the `createFSEvent` method name.  I'd leave it up to the developer to `event` or `log` using respective methods.  Or is there some greater use-case this solves?